internal/controller/progressive/progressive.go

```diff
@@ -171,12 +167,6 @@ func ProcessResource(
 
 	// There's already an Upgrading child, now process it
 
-	// Get the live resource so we don't have issues with an outdated cache
-	currentUpgradingChildDef, err = kubernetes.GetLiveResource(ctx, currentUpgradingChildDef, rolloutObject.GetChildGVR().Resource)
-	if err != nil {
-		return false, 0, err
-	}
-
```

Comment:
up on line 169 I think is the only place where we were not calling the live K8S API, and really we don't hit that line very often